.github/workflows/test-gutenberg-build-process-run.yml

```diff
@@ -0,0 +1,100 @@
+##
+# A callable workflow that tests the Gutenberg plugin build process.
+##
+name: Test the Gutenberg plugin Build Process
+
+on:
+  workflow_call:
+    inputs:
+      os:
+        description: 'Operating system to run tests on'
+        required: false
+        type: 'string'
+        default: 'ubuntu-latest'
+      with-core:
+        description: 'Whether to test with WordPress Core'
+        required: false
+        type: 'boolean'
+        default: false
+      directory:
+        description: 'Directory to run WordPress out of. Valid values are src or build'
+        required: false
+        type: 'string'
+        default: 'src'
+
+env:
+  GUTENBERG_DIRECTORY: ${{ inputs.directory == 'build' && 'build' || 'src' }}/wp-content/plugins/gutenberg
+  PUPPETEER_SKIP_DOWNLOAD: ${{ true }}
+
+jobs:
+  # Verifies that installing npm dependencies and building the Gutenberg plugin works as expected.
+  #
+  # Performs the following steps:
+  # - Checks out the repository.
+  # - Checks out the Gutenberg plugin repository in the plugins directory.
+  # - Sets up Node.js.
+  # - Logs debug information about the GitHub Action runner.
+  # - Installs Core npm dependencies if testing with Core.
+  # - Installs npm dependencies for Gutenberg.
+  # - Runs the Gutenberg build process.
+  # - Builds WordPress to run from the relevant location (src or build) if testing with Core.
+  # - Runs the Gutenberg build process.
+  # - Ensures version-controlled files are not modified or deleted.
+  build-process-tests:
```

Comment:
`with-core: false` seems pointless. It's not our job to test whether Gutenberg can simply build by itself. That's supposed to be tested in the Gutenberg repository itself.

Reply:
That's a good point. Thinking back, I guess my intention here was to have a job that acts as a control that confirms the build process works on its own when there are errors in the other ones.

Reply:
Removed this for now. If we're finding it hard to debug failures, we can look at adding this back.